### PR TITLE
Implemented basic server

### PR DESCRIPTION
**Pull request by Ahmed-Elawad**
Defined server directory and implemented basic server with response to standard GET request.
- uses express native body parsing
- implemented routing for 'product/:id' endpoint to provide json response of all product details

TODO(Next steps):
- defined public directory for static files
- Implement full test suite for unit test of a GET req and it's res obj

### CODE REVIEW

**Comment by hongmm** [comment]
@Ahmed-Elawad API code LGTM! merge away unless you want another pair of eyes on this. Code looks fine and I assume you've manually tested/created a test for this

**Review by augustdolan — COMMENTED** [review]
Looking good Ahmed! A few things:

- Don't feel a need to say what you use in your code on the pull request. Folks will know you use express.json() by reviewing your code. Keep the explanations to 'what' instead of 'how,' and let all your good coding do the rest of the explanation for you.

- You can gather id off of params directly via req.params.id, no need to pull the object out of your req.

- Make sure you put parens around your (resp) => {} to sync up with HR style guides.

**Comment by Ahmed-Elawad** [comment]
@augustdolan, Thanks for the notes! I'll get a ticket in to correct those.

**Comment by Ahmed-Elawad** [comment]
@hongmm, I'm finalizing my testing suite for it today hopefully as it's own feature.

File: server/index.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+const express = require('express');
+const app = express();
+const db = require('../database/index.js');
+
+app.use(express.json());
+app.use(express.urlencoded({extended: true}));
+// app.use(express.static());
```
Comment (jonathandavidlewis) [inline server/index.js:7]: Extra comment.

File: server/index.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    .then((resp) => {
+        res.json(resp)
+    })
+    .catch((err) => {
+        res.end(err);
+    })
+})
+
+const port = 3000;
+app.listen(port, () => console.log(`Listening on port:${port}`));
```
Comment (jonathandavidlewis) [inline server/index.js:20]: Please add a blank line at the EOF, end of all js files: 
https://stackoverflow.com/questions/2287967/why-is-it-recommended-to-have-empty-line-in-the-end-of-a-source-file

File: server/index.js (hunk 1 above)
Comment (Ahmed-Elawad) [reply server/index.js]: Thanks for the catch!

File: server/index.js (hunk 2 above)
Comment (Ahmed-Elawad) [reply server/index.js]: All set on the EOF lines.

**Pull request merged**